src/core/space_interaction_API.jl

```diff
@@ -169,6 +198,12 @@ function genocide!(model::ABM, n::Integer)
     model.maxid[] = n
 end
 
+# Must be implemented separately, otherwise genocide! would return an unhelpful
+# BoundsError error to the user.
+function genocide!(model::ABM{S,A,Vector{A},F,P,R}, n::Integer) where {S,A,F,P,R}
+    nokill()
+end
+
```

Comment:
Same as above; as long as `remove_agent_from_model!` does what it is supposed to be, all these downstream functions don't need special methods. The function `nokill()` itself has no reason to exist. `remove_agent_from_model!` with `SizedArray` should become `nokill()`.

Reply:
This one I implemented specifically because iteration in the above one uses `for (k, v) in model.agents` which would throw a bounds error for vector-stored agents rather than anything about not being able to remove agents. This implementation is not perfect, but it saves having to qualify the above `genocide!` function with `genocide!(model::ABM{..., Dict{Int, A}} where ...`.

Reply:
On a slightly wider note: I think all of the arguments to genocide besides `model` should be keyword only. This comes about because `genocide!(model, [5])` and `genocide!(model, 5)` do something so dramatically different, the prior of which I wouldn't even call "genocide".